extends Node2D

@onready var path = $Path2D
@onready var path_follower = $Path2D/PathFollower
@onready var player = $Path2D/PathFollower/Employeur
@onready var animated_sprite = $Path2D/PathFollower/Employeur/AnimatedSprite2D
@onready var porte = $porte/AnimatedSprite2D
@onready var sign = $Sign/AnimatedSprite2D

func _ready():

	print("Appel de place_player_at_last_offset, last_clicked_object: ", get_node("/root/GameManager").last_clicked_object)
	get_node("/root/GameManager").place_player_at_last_offset()
	if GameManager and GameManager.last_clicked_object != "":
		await get_tree().process_frame
		GameManager.place_player_at_last_offset()
		print('place_player_at_last_offest est appelÃ©')
	else:
		print("â© Aucun objet encore cliquÃ©, le joueur ne sera pas dÃ©placÃ© au dÃ©marrage.")
		GameManager.init_refs(self)
		path_follower.progress_ratio = GameManager.player_path_ratio
		await get_tree().process_frame
		#GameManager.place_player_at_last_offset()
	if player and GameManager:
		print("âœ… ScÃ¨ne chez_yann prÃªte avec Employeur et GameManager")
	else:
		push_error("âŒ Erreur : Employeur ou GameManager non trouvÃ© dans chez_yann.tscn")	
		
	if GameManager:
		GameManager.reset_bubbles()
		# ğŸ”„ Reset complet des rÃ©fÃ©rences
	if GameManager:
		# ğŸ”„ Reset des bulles
		GameManager.current_bubble = null
		
		# ğŸ‘‡ RÃ©assigne le container Ã  chaque retour
		
		var bubble_container = $UI/SpeechBubbleContainer
		if bubble_container:
			GameManager.speech_bubble_container = bubble_container
		else:
			push_error("âš ï¸ Aucun SpeechBubbleContainer trouvÃ© dans chez_yann.tscn")

	animated_sprite.play('front')
	
		# ğŸ‘‡ Tous les objets visitÃ©s
	if GameManager.all_objects_visited():
		if not GameManager.door_opened:
			# â³ Petite pause avant la premiÃ¨re ouverture
			await get_tree().create_timer(1.0).timeout
			porte.play("anim")
			sign.play('on')
			GameManager.door_opened = true
		else:
			# ğŸšª DÃ©jÃ  ouverte â†’ on affiche directement lâ€™Ã©tat final
			porte.play("open")
			sign.play('on')
			porte.frame = porte.sprite_frames.get_frame_count("open") - 1
	else:
		# ğŸ”’ Encore fermÃ©e
		porte.play("close")
		sign.play('off')
	
	
